host/class/uac/usb_host_uac/uac_host.c

```diff
@@ -0,0 +1,2141 @@
+                iface_alt->ep_attr = ep_desc->bmAttributes;
+                iface_alt->interval = ep_desc->bInterval;
+                ESP_LOGD(TAG, "UAC Endpoint 0x%02X, Max Packet Size %d, Attributes 0x%02X, Interval %d", ep_desc->bEndpointAddress, ep_desc->wMaxPacketSize, ep_desc->bmAttributes, ep_desc->bInterval);
+                break;
+            }
+            case UAC_CS_ENDPOINT: {
+                // we has got enough information to fill the uac_iface_alt_t, so we can break
+                const uac_as_cs_ep_desc_t *cs_ep_desc = (const uac_as_cs_ep_desc_t *)cs_desc;
+                if (cs_ep_desc->bDescriptorSubtype == UAC_EP_GENERAL) {
+                    iface_alt->freq_ctrl_supported = cs_ep_desc->bmAttributes & UAC_SAMPLING_FREQ_CONTROL;
+                    parse_continue = false;
+                    ESP_LOGD(TAG, "UAC EP General, Attributes 0x%02X", cs_ep_desc->bmAttributes);
+                    ESP_LOGD(TAG, "UAC EP Frequency Control %d", iface_alt->freq_ctrl_supported);
+                }
+                break;
+            }
+            default:
+                break;
+            }
+            cs_desc = GET_NEXT_DESC(cs_desc, total_length, cs_offset);
+        }
+        // Get next alternate setting
+        iface_alt_desc = GET_NEXT_INTERFACE_DESC(iface_alt_desc, total_length, iface_alt_offset);
+    }
+    uac_iface->state = UAC_INTERFACE_STATE_NOT_INITIALIZED;
+    uac_iface->parent = uac_device;
+    uac_iface->dev_info.addr = uac_device->addr;
+    uac_iface->dev_info.iface_num = iface_desc->bInterfaceNumber;
+    uac_iface->dev_info.iface_alt_num = iface_alt_idx;
+    ESP_LOGD(TAG, "UAC Interface %d, found total alternate %d", iface_desc->bInterfaceNumber, iface_alt_idx);
+
+    // Fill descriptor device information
+    const usb_device_desc_t *desc;
+    usb_device_info_t dev_info;
+    UAC_GOTO_ON_ERROR(usb_host_get_device_descriptor(uac_device->dev_hdl, &desc), "Unable to get device descriptor");
+    UAC_GOTO_ON_ERROR(usb_host_device_info(uac_device->dev_hdl, &dev_info), "Unable to get USB device info");
+    // VID, PID
+    uac_iface->dev_info.VID = desc->idVendor;
+    uac_iface->dev_info.PID = desc->idProduct;
+    // Strings
+    uac_host_string_descriptor_copy(uac_iface->dev_info.iManufacturer, dev_info.str_desc_manufacturer);
+    uac_host_string_descriptor_copy(uac_iface->dev_info.iProduct, dev_info.str_desc_product);
+    uac_host_string_descriptor_copy(uac_iface->dev_info.iSerialNumber, dev_info.str_desc_serial_num);
+
+    *p_uac_iface = uac_iface;
+
+    UAC_ENTER_CRITICAL();
+    STAILQ_INSERT_TAIL(&s_uac_driver->uac_ifaces_tailq, uac_iface, tailq_entry);
+    UAC_EXIT_CRITICAL();
+
+    return ESP_OK;
+
+fail:
+    if (uac_iface && uac_iface->state_mutex) {
+        vSemaphoreDelete(uac_iface->state_mutex);
+    }
+    free(uac_iface->iface_alt);
+    free(uac_iface);
+    return ret;
+}
+
+/**
+ * @brief Remove interface from a list
+ *
+ * Use only inside critical section
```

Comment:
Outdated comment? the function enters critical section so this would cause a deadlock